### ISSUE TITLE
🔄 Regression on importing from subdirectory of a package in node_modules

### ISSUE THREAD

**Issue by stephenwade** [webpro-nl/knip#773]
### Prerequisites

- [X] I've read the relevant [documentation](https://knip.dev)
- [X] I've searched for [existing issues](https://github.com/webpro-nl/knip/issues?q=is%3Aissue)
- [X] I've read the [issue reproduction guide](https://knip.dev/guides/issue-reproduction)

### Reproduction url

https://github.com/stephenwade/knip-issue-repro-crispy-broccoli

### Reproduction access

- [X] I've made sure the reproduction is publicly accessible

### Good version

5.27.3

### Bad version

5.27.4

### Description of the regression

In my project, I import a type from `'node_modules/react-toastify/dist/components'` because `'react-toastify'` doesn't export it. To get around Knip reporting it as an unlisted dependency, I added this to my config file:

```
"ignoreDependencies": ["node_modules/react-toastify/dist/components"]
```

In version 5.27.3, Knip reports no issues. In version 5.27.4 and later, I get this report:

```
Unlisted dependencies (1)
node_modules/react-toastify/dist/components  app/ToastContainer.tsx
Configuration issues (1)
Unused item in ignoreDependencies: node_modules/react-toastify/dist/components
```

I haven't found a configuration that works for my project since the update.

I think the issue is likely caused by 89780376.

(ref: https://github.com/stephenwade/festival/pull/1693)

**Comment by webpro** [comment]
Regardless of whether this is a bug or not, this seems to work the same for a while:

```
npx knip@5.0.0
Unlisted dependencies (1)
node_modules/react-toastify/dist/components  app/ToastContainer.tsx
```

The values in `ignoreDependencies` are package names so either `"name"` or `"@scope/name"` - so in this case the odd `"node_modules"` would do.

Will investigate some more later on.

**Comment by stephenwade** [comment]
The issue is not that it appears as an unlisted dependency. The issue is that I can no longer ignore that report with `"ignoreDependencies"`.

**Comment by webpro** [comment]
You can use literally `"ignoreDependencies": ["node_modules"]`

**Comment by stephenwade** [comment]
That works, thank you!

I'll leave the issue open for the underlying issue that Knip should detect this import as `react-toastify` and not need any `"ignoreDependencies"` config.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.30.2. See [Release 5.30.2](https://github.com/webpro-nl/knip/releases/tag/5.30.2) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
